src/Bicep.Core/Parser/Parser.cs

```diff
@@ -670,6 +666,20 @@ private SyntaxBase WithRecovery<TSyntax>(Func<TSyntax> syntaxFunc, bool consumeT
             }
         }
 
+        private SyntaxBase? WithRecoveryNullable<TSyntax>(Func<TSyntax> syntaxFunc, bool consumeTerminator, params TokenType[] terminatingTypes)
+            where TSyntax : SyntaxBase?
```

Comment:
I don't think it matters that much, but I'm more used to seeing e.g. `Func<TSyntax?>` and `where TSyntax : SyntaxBase`. Feels like it would be better to consider `TSyntax` non-nullable.

Reply:
Left as-is for now...